### ISSUE TITLE
Source filtering

### ISSUE THREAD

**Issue by roberth** [srid/haskell-flake#13]
Changes to unrelated files cause rebuilds, especially when the package is at the root of the repo, where non-package files have no other place to go.

Some solutions:
a. parse the cabal file in Nix: no experimental features required, but somewhat hard to get right, not future proof
b. or use ca-derivations with buildFromSdist or buildFromCabalSdist  (https://github.com/NixOS/nixpkgs/pull/174176)

**Comment by srid** [comment]
This can also do this, right? https://github.com/numtide/nix-filter

One would have manually set up the filters, however.

**Comment by roberth** [comment]
Yeah, a basic solution is to make the src configurable for the package. (or packages #7 ) Users can then make use of any filtering library they like (`lib`, nix-filter, [`lib.sources` combinators](https://github.com/NixOS/nixpkgs/pull/112083) (a bit heavy as a flake though), ...).

**Comment by srid** [comment]
@roberth Is this still relevant?

**Comment by roberth** [comment]
While technically the automated filtering can still be improved, I don't think it's significant or important enough.
It's best to put a haskell package in its own directory anyway, in which case it would seem unlikely that you'd have any significant amount of redundant files in such a directory.
And if you do, the usual source filtering can be done by setting `packages.<name>.root = someLib.someFunction ./<name>`. Is that documented?

**Comment by ParetoOptimalDev** [comment]
> It's best to put a haskell package in its own directory anyway,

Does this mean something like don't let `Module.hs` and `.docker` be in the same directory?

**Comment by srid** [comment]
https://community.flake.parts/haskell-flake/local#source-filtering
